dashboard/app/controllers/lti_v1_controller.rb

```diff
@@ -42,10 +42,45 @@ def login
     redirect_to auth_redirect_url.to_s
   end
 
+  def authenticate
+    id_token = params[:id_token]
+    return unauthorized_status unless id_token
+    begin
+      decoded_jwt_no_auth = JSON::JWT.decode(id_token, :skip_verification)
+    rescue
+      return unauthorized_status
+    end
+    # client_id is the aud[ience] in the JWT
+    extracted_client_id = decoded_jwt_no_auth[:aud]
```

Comment:
Now I'm re-reading the [LTI Security Spec](https://www.imsglobal.org/spec/security/v1p0/#id-token) on the `id_token`. Looks like the `aud` key can be an array, so this `extracted_client_id` could fail in that case when we use it in the db query later on line 57 right?

Reply:
Yeah, that's why I think the other example puts it all in the jwt_verifier and that works with an array.. but also the test that I have `auth - given a valid jwt with the audience as an array, redirect to target_link_url` does successfully redirect. I don't know why 🤷

Reply:
Ah! I figured it out.

* `find_by` is just a wrapper for `where` - [link](https://github.com/rails/rails/blob/8015c2c2cf5c8718449677570f372ceb01318a32/activerecord/lib/active_record/relation/finder_methods.rb#L80)
* `where` is pretty intelligent with the query options you can pass it. It can take a hash, where a key's value is an array - [link](https://apidock.com/rails/ActiveRecord/QueryMethods/where).
Excerpt from the link
```
User.where({ name: ["Alice", "Bob"]})
# SELECT * FROM users WHERE name IN ('Alice', 'Bob')
```